Index out-of-bounds error in terraform provider caused by adding clients to an auth selector

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### PingFederate Terraform provider Version
v1.7.1

### PingFederate Version
12.2.3

### Terraform Version
OpenTofu v1.11.5

### Affected Resource(s)
- pingfederate_authentication_selector

### Terraform Configuration Files
```hcl
resource "pingfederate_authentication_selector" "test" {
  attribute_contract = {
    extended_attributes = null
  }
  configuration = {
    fields = [
    ]
    sensitive_fields = [
    ]
    tables = [
      {
        name = "Clients"
        rows = [
          {
            default_row = false
            fields = [
              {
                name  = "Client ID"
                value = "foo"
              },
            ]
            sensitive_fields = [
            ]
          },
        ]
      },
    ]
  }
  name       = "Test"
  parent_ref = null
  plugin_descriptor_ref = {
    id = "com.pingidentity.pf.selectors.oauthclient.OAuthClientAdapterSelector"
  }
  selector_id = "Test"
}
```

### Debug Output
See panic below:  Can provide full debug if required

### Panic Output
```
2026-04-24T13:39:49.174-0500 [DEBUG] ReferenceTransformer: "pingfederate_authentication_selector.test" references: []
2026-04-24T13:39:49.491-0500 [ERROR] provider.terraform-provider-pingfederate_v1.7.1.exe: panic: runtime error: index out of range [1] with length 1
2026-04-24T13:39:49.491-0500 [ERROR] provider.terraform-provider-pingfederate_v1.7.1.exe
2026-04-24T13:39:49.491-0500 [ERROR] provider.terraform-provider-pingfederate_v1.7.1.exe: goroutine 27 [running]:
2026-04-24T13:39:49.491-0500 [ERROR] provider.terraform-provider-pingfederate_v1.7.1.exe: github.com/pingidentity/terraform-provider-pingfederate/internal/resource/common/pluginconfiguration.readRowsResponse({0xc0003bc300, 0x2, 0x2?}, 0x2?, 0xc0006b6a78)
2026-04-24T13:39:49.491-0500 [ERROR] provider.terraform-provider-pingfederate_v1.7.1.exe: 	github.com/pingidentity/terraform-provider-pingfederate/internal/resource/common/pluginconfiguration/plugin_configuration_state.go:257 +0x10ad
2026-04-24T13:39:49.491-0500 [ERROR] provider.terraform-provider-pingfederate_v1.7.1.exe: github.com/pingidentity/terraform-provider-pingfederate/internal/resource/common/pluginconfiguration.toTablesSetValue({0xc0004808a0, 0x1, 0x0?}, 0xc0006b6928, 0xc0006b6a78)
2026-04-24T13:39:49.491-0500 [ERROR] provider.terraform-provider-pingfederate_v1.7.1.exe: 	github.com/pingidentity/terraform-provider-pingfederate/internal/resource/common/pluginconfiguration/plugin_configuration_state.go:332 +0x5a8
2026-04-24T13:39:49.491-0500 [ERROR] provider.terraform-provider-pingfederate_v1.7.1.exe: github.com/pingidentity/terraform-provider-pingfederate/internal/resource/common/pluginconfiguration.ToState({0xc0003928d0?, 0xc0003585a0?, 0xd0?}, 0xc0004b6038, 0x0)
2026-04-24T13:39:49.491-0500 [ERROR] provider.terraform-provider-pingfederate_v1.7.1.exe: 	github.com/pingidentity/terraform-provider-pingfederate/internal/resource/common/pluginconfiguration/plugin_configuration_state.go:389 +0x7bb
2026-04-24T13:39:49.491-0500 [ERROR] provider.terraform-provider-pingfederate_v1.7.1.exe: github.com/pingidentity/terraform-provider-pingfederate/internal/resource/config/authenticationselector.readAuthenticationSelectorsResponse({0x7ff6ed90e558, 0xc0003c6b10}, 0xc0004b6000, 0xc00013e420, {0xc0003928d0?, 0xc0003585a0?, 0x0?}, 0x0)
2026-04-24T13:39:49.491-0500 [ERROR] provider.terraform-provider-pingfederate_v1.7.1.exe: 	github.com/pingidentity/terraform-provider-pingfederate/internal/resource/config/authenticationselector/authentication_selector_resource.go:201 +0x37d
2026-04-24T13:39:49.491-0500 [ERROR] provider.terraform-provider-pingfederate_v1.7.1.exe: github.com/pingidentity/terraform-provider-pingfederate/internal/resource/config/authenticationselector.(*authenticationSelectorResource).Read(0xc0003c2800, {0x7ff6ed90e558, 0xc0003c6b10}, {{{{0x7ff6ed915090, 0xc00033ebd0}, {0x7ff6ed49c5c0, 0xc000339380}}, {0x7ff6ed925978, 0xc0000f1ae0}}, 0x0, ...}, ...)
2026-04-24T13:39:49.491-0500 [ERROR] provider.terraform-provider-pingfederate_v1.7.1.exe: 	github.com/pingidentity/terraform-provider-pingfederate/internal/resource/config/authenticationselector/authentication_selector_resource.go:279 +0x38f
2026-04-24T13:39:49.491-0500 [ERROR] provider.terraform-provider-pingfederate_v1.7.1.exe: github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ReadResource(0xc000144a88, {0x7ff6ed90e558, 0xc0003c6b10}, 0xc00062a000, 0xc0006b7640)
2026-04-24T13:39:49.491-0500 [ERROR] provider.terraform-provider-pingfederate_v1.7.1.exe: 	github.com/hashicorp/terraform-plugin-framework@v1.19.0/internal/fwserver/server_readresource.go:156 +0xb8e
2026-04-24T13:39:49.491-0500 [ERROR] provider.terraform-provider-pingfederate_v1.7.1.exe: github.com/hashicorp/terraform-plugin-framework/internal/proto6server.(*Server).ReadResource(0xc000144a88, {0x7ff6ed90e558?, 0xc0003c6a20?}, 0xc000134500)
2026-04-24T13:39:49.491-0500 [ERROR] provider.terraform-provider-pingfederate_v1.7.1.exe: 	github.com/hashicorp/terraform-plugin-framework@v1.19.0/internal/proto6server/server_readresource.go:71 +0x592
2026-04-24T13:39:49.491-0500 [ERROR] provider.terraform-provider-pingfederate_v1.7.1.exe: github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ReadResource(0xc000234460, {0x7ff6ed90e558?, 0xc000219e60?}, 0xc000442230)
2026-04-24T13:39:49.491-0500 [ERROR] provider.terraform-provider-pingfederate_v1.7.1.exe: 	github.com/hashicorp/terraform-plugin-go@v0.31.0/tfprotov6/tf6server/server.go:864 +0x2f3
2026-04-24T13:39:49.491-0500 [ERROR] provider.terraform-provider-pingfederate_v1.7.1.exe: github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ReadResource_Handler({0x7ff6ed635640, 0xc000234460}, {0x7ff6ed90e558, 0xc000219e60}, 0xc0003c2000, 0x0)
2026-04-24T13:39:49.491-0500 [ERROR] provider.terraform-provider-pingfederate_v1.7.1.exe: 	github.com/hashicorp/terraform-plugin-go@v0.31.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:925 +0x1a9
2026-04-24T13:39:49.491-0500 [ERROR] provider.terraform-provider-pingfederate_v1.7.1.exe: google.golang.org/grpc.(*Server).processUnaryRPC(0xc000222d88, {0x7ff6ed90e558, 0xc000219da0}, 0xc000630000, 0xc0003426c0, 0x7ff6ee3ee3d8, 0x0)
2026-04-24T13:39:49.491-0500 [ERROR] provider.terraform-provider-pingfederate_v1.7.1.exe: 	google.golang.org/grpc@v1.79.3/server.go:1429 +0x1192
2026-04-24T13:39:49.492-0500 [ERROR] provider.terraform-provider-pingfederate_v1.7.1.exe: google.golang.org/grpc.(*Server).handleStream(0xc000222d88, {0x7ff6ed90f138, 0xc0001629c0}, 0xc000630000)
2026-04-24T13:39:49.492-0500 [ERROR] provider.terraform-provider-pingfederate_v1.7.1.exe: 	google.golang.org/grpc@v1.79.3/server.go:1855 +0xbc6
2026-04-24T13:39:49.492-0500 [ERROR] provider.terraform-provider-pingfederate_v1.7.1.exe: google.golang.org/grpc.(*Server).serveStreams.func2.1()
2026-04-24T13:39:49.492-0500 [ERROR] provider.terraform-provider-pingfederate_v1.7.1.exe: 	google.golang.org/grpc@v1.79.3/server.go:1064 +0x7f
2026-04-24T13:39:49.492-0500 [ERROR] provider.terraform-provider-pingfederate_v1.7.1.exe: created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 26
2026-04-24T13:39:49.492-0500 [ERROR] provider.terraform-provider-pingfederate_v1.7.1.exe: 	google.golang.org/grpc@v1.79.3/server.go:1075 +0x11d
```

### Expected Behavior
Terraform should have created a plan for the Authentication Selector (in this case, setting it back to a single client in the table

### Actual Behavior
During planning there is an index out of bounds error, then🔥 PANIC 🔥

### Steps to Reproduce
1. Use Terraform to create a new authentication selector resource.  In this case I used a OAuthClientAdapterSelector, but in production the error has occured with SAML selectors as well.
2. In the Ping Fed console, add another client to the auth selector
3. Back in Terraform, re-run "tofu plan"
4. Watch the debug logs report that there's an index issue trying to read a list value that is one past the end of the list.  (My assumption is that it is trying to read the current state based off the index of the updated state, which has an extra member, so the list length is one more)

For additional information, in production I'm trying to create authentication selectors with the lifecycle attribute ignore_changes = all.  That way I can create the authentication selector with a single client, but have day-to-day operations add or remove clients via the GUI.  This worked until a new client was added to a selector, at which time we got the behavior described above.

### References
None that I know of

--- a/.changelog/pr-603.txt
+++ b/.changelog/pr-603.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+Fixed an issue where adding table rows to plugin configuration outside of terraform (such as via the PingFederate UI) could cause a panic when running a subsequent plan.
+```

--- a/internal/acctest/config/passwordcredentialvalidator/password_credential_validator_table_rows_drift_test.go
+++ b/internal/acctest/config/passwordcredentialvalidator/password_credential_validator_table_rows_drift_test.go
@@ -1,0 +1,76 @@
+// Copyright © 2026 Ping Identity Corporation
+
+package passwordcredentialvalidator_test
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-framework/providerserver"
+	"github.com/hashicorp/terraform-plugin-go/tfprotov6"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	client "github.com/pingidentity/pingfederate-go-client/v1300/configurationapi"
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/acctest"
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/provider"
+)
+
+// This test validates that if a row is added to a plugin configuration table outside of Terraform,
+// a non-empty plan is generated reflecting the drift, with no panic or error.
+func TestAccPasswordCredentialValidator_RadiusTableRowsDrift(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() { acctest.ConfigurationPreCheck(t) },
+		ProtoV6ProviderFactories: map[string]func() (tfprotov6.ProviderServer, error){
+			"pingfederate": providerserver.NewProtocol6WithError(provider.NewTestProvider()),
+		},
+		CheckDestroy: passwordCredentialValidator_CheckDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: passwordCredentialValidator_RadiusCompleteHCL(),
+			},
+			{
+				PreConfig: func() {
+					passwordCredentialValidator_AddRadiusServerRow(t)
+				},
+				RefreshState:       true,
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func passwordCredentialValidator_AddRadiusServerRow(t *testing.T) {
+	testClient := acctest.TestClient()
+	ctx := acctest.TestBasicAuthContext()
+
+	validator, _, err := testClient.PasswordCredentialValidatorsAPI.GetPasswordCredentialValidator(ctx, validatorId).Execute()
+	if err != nil {
+		t.Fatalf("Failed to get password credential validator: %v", err)
+	}
+
+	tableIndex := -1
+	for i, table := range validator.Configuration.Tables {
+		if table.Name == "RADIUS Servers" {
+			tableIndex = i
+			break
+		}
+	}
+
+	if tableIndex < 0 {
+		t.Fatal("Failed to find RADIUS Servers table in password credential validator configuration")
+	}
+
+	newRow := client.ConfigRow{Fields: []client.ConfigField{
+		{Name: "Hostname", Value: client.PtrString("localhost2")},
+		{Name: "Authentication Port", Value: client.PtrString("1812")},
+		{Name: "Authentication Protocol", Value: client.PtrString("PAP")},
+		{Name: "Shared Secret", Value: client.PtrString("mysecret2")},
+	}}
+
+	validator.Configuration.Tables[tableIndex].Rows = append(validator.Configuration.Tables[tableIndex].Rows, newRow)
+
+	updateRequest := testClient.PasswordCredentialValidatorsAPI.UpdatePasswordCredentialValidator(ctx, validatorId)
+	updateRequest = updateRequest.Body(*validator)
+	_, _, err = testClient.PasswordCredentialValidatorsAPI.UpdatePasswordCredentialValidatorExecute(updateRequest)
+	if err != nil {
+		t.Fatalf("Failed to update password credential validator with additional row: %v", err)
+	}
+}

--- a/internal/resource/common/pluginconfiguration/plugin_configuration_state.go
+++ b/internal/resource/common/pluginconfiguration/plugin_configuration_state.go
@@ -245,32 +245,37 @@ func readRowsResponse(rows []client.ConfigRow, planRows *types.List, diags *diag
 			rowsSensitiveFieldsSplit = append(rowsSensitiveFieldsSplit, rowSensitiveFieldsSplit)
 		}
 	} else {
-		// This is assuming there are never any rows added by the PF API. If there
-		// are ever rows added, this will cause a nil pointer exception trying to read
-		// index i of planRowsElements.
 		planRowsElements := planRows.Elements()
 		for i := 0; i < len(rows); i++ {
 			attrValues := map[string]attr.Value{}
 			attrValuesSensitiveSplit := map[string]attr.Value{}
 			attrValues["default_row"] = types.BoolPointerValue(rows[i].DefaultRow)
 			attrValuesSensitiveSplit["default_row"] = types.BoolPointerValue(rows[i].DefaultRow)
-			planRow := planRowsElements[i].(types.Object)
+			rowInPlan := i < len(planRowsElements)
 			var planRowFields, planRowSensitiveFields *types.Set
-			planRowFieldsVal, ok := planRow.Attributes()["fields"]
-			if ok {
-				setVal := planRowFieldsVal.(types.Set)
-				planRowFields = &setVal
-			}
-			planRowSensitiveFieldsVal, ok := planRow.Attributes()["sensitive_fields"]
-			if ok {
-				setVal := planRowSensitiveFieldsVal.(types.Set)
-				planRowSensitiveFields = &setVal
+			if rowInPlan {
+				planRow := planRowsElements[i].(types.Object)
+				planRowFieldsVal, ok := planRow.Attributes()["fields"]
+				if ok {
+					setVal := planRowFieldsVal.(types.Set)
+					planRowFields = &setVal
+				}
+				planRowSensitiveFieldsVal, ok := planRow.Attributes()["sensitive_fields"]
+				if ok {
+					setVal := planRowSensitiveFieldsVal.(types.Set)
+					planRowSensitiveFields = &setVal
+				}
 			}
 
 			rowFields := readFieldsResponse(rows[i].Fields, planRowFields, planRowSensitiveFields, diags)
 			attrValues["fields"] = rowFields.allFields
-			attrValuesSensitiveSplit["fields"] = rowFields.plannedCleartextFields
-			attrValuesSensitiveSplit["sensitive_fields"] = rowFields.plannedSensitiveFields
+			if rowInPlan {
+				attrValuesSensitiveSplit["fields"] = rowFields.plannedCleartextFields
+				attrValuesSensitiveSplit["sensitive_fields"] = rowFields.plannedSensitiveFields
+			} else {
+				attrValuesSensitiveSplit["fields"] = rowFields.allCleartextFields
+				attrValuesSensitiveSplit["sensitive_fields"] = rowFields.allSensitiveFields
+			}
 
 			rowMergedFields, respDiags := types.ObjectValue(rowsMergedFieldsAttrTypes, attrValues)
 			diags.Append(respDiags...)

